src/pages/EnvironmentsPage/environmentsPermissions.ts

```diff
@@ -0,0 +1,26 @@
+import { LeoResourcePermissionsProvider } from 'src/analysis/Environments/Environments';
+import { getCreatorForCompute } from 'src/analysis/utils/resource-utils';
+import { IHaveCreator } from 'src/libs/ajax/leonardo/models/core-models';
+import { getTerraUser } from 'src/libs/state';
+
+const makePermissionsProvider = (userEmailGetter: () => string): LeoResourcePermissionsProvider => {
```

Comment:
yes, it's an arrow function so that the current value is always retrieved on execution of the permission methods.

The goal is to remove the concern of permissions from the already large Environments component, and also letting AoU determine this logic, and where things like current user email state live, according to their own needs.  Also Justin's comment above highlighted how moving this concern out of the component is a nice overall direction for future needs.

Reply:
Why would the user email change in between invocations of permission functions?

Reply:
In which case, they'd be redirected away from this page.

Reply:
basically, it's backed by state, which means it can change, so for defensiveness, I made it an arrow function....